### ISSUE TITLE
More flexible pinning of JupyterLab, Jupyter Notebook, IPython, and lower level components

### ISSUE THREAD

**Issue by astrojuanlu** [kedro-org/kedro#2700]
## Description
We're pinning Jupyter in various ways in the development requirements for projects:

https://github.com/kedro-org/kedro/blob/c48a8469b0fda3a39690c58de95b9f90165a0bc6/kedro/templates/project/%7B%7B%20cookiecutter.repo_name%20%7D%7D/src/requirements.txt#L4-L8

and we're pulling more dependencies than needed. For example, `jupyter` in itself is basically a metapackage:

https://pypi.org/project/jupyter/1.0.0/#files

and it ships a bunch of stuff:

```
    install_requires    = [
        'notebook',
        'qtconsole',
        'jupyter-console',
        'nbconvert',
        'ipykernel',
        'ipywidgets',
    ],
```

I don't think we want to necessarily pull things like `qtconsole` or even `ipywidgets`.

This is related to https://github.com/kedro-org/kedro/issues/2519 in the sense that these should not be project dependencies anyway, but even when we address gh-2519, we should make these dependencies slimmer (see also (see https://github.com/kedro-org/kedro/pull/2569#discussion_r1202077782 for extra discussion). At the same time, our `kedro.ipython` extension isn't doing anything fancy, so apart from incompatibilities with other packages, we should keep upper version caps to a minimum I'd say.

## Context
I'm happy to report that `%load_ext kedro.ipython` works nicely in Jupyter (Classic) Notebook 7.0.0rc0 🎉 

![image](https://github.com/kedro-org/kedro/assets/316517/480b743d-f795-4c7f-b446-9f63a9547067)

And it would be cool if our users could benefit from the new interface as soon as it's out.

Note that Notebook is now based on JupyterLab, so it has _more_ dependencies.

Anecdotally, I've found that many users still stick to the classic notebook interface and not all of them have migrated, nor intend to, to JupyterLab.

## Possible Implementation
Make our pinnings something like

```
# requirements.txt
ipython>=8.10  # runtime dependency of kedro.ipython
notebook>=7.0  # favor modern version for better usability
# jupyterlab  # it's a dependency of notebook already, so not even needed
```

This would be blocked on gh-2158.

## Possible Alternatives
Alternatively, we could drop the classic notebook and have users install only JupyterLab:

```
# requirements.txt
ipython>=8.10  # runtime dependency of kedro.ipython
jupyterlab>=3.0  # whatever version, old or new
# notebook  # not included, fewer dependencies are installed
```

**Comment by deepyaman** [comment]
Would this PR include evaluation of Jupiter Lab 4.x? I don't think anything should break, but it's probably worth checking in relaxing it. Noting, as discussed in backlog grooming, so it's not forgotten.

**Comment by astrojuanlu** [comment]
Jupyter Notebook 7.0.0 is out https://github.com/jupyter/notebook/releases/tag/v7.0.0

**Comment by astrojuanlu** [comment]
Closing this in favour of #2276.
